### PR TITLE
Add marker error type and wrap error on not found marker

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

Currently, when querying for a denom by address or name we get a detailed response if it isn't found. invalid marker denom: pool.47qvz2sfcj1d76kr9plu3y: decoding bech32 failed: checksum failed. Expected zmz37k, got 9plu3y.

The response can be simplified to marker not found; invalid denom or address.

closes: #325

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/marker/keeper/querier.go [hunk 1]
```diff
@@ -165,11 +165,11 @@ func accountForDenomOrAddress(ctx sdk.Context, keeper Keeper, lookup string) (ty
 	if addr, addrErr = sdk.AccAddressFromBech32(lookup); addrErr != nil {
 		account, err = keeper.GetMarkerByDenom(ctx, lookup)
 		// if we error on get by denom, ensure we also capture the fail to parse as an address
```
Comment (iramiller) [inline x/marker/keeper/querier.go:167]: this comment doesn't apply anymore

**Review by iramiller — COMMENTED** [review]
minor nit.

File: x/marker/keeper/querier.go (hunk 1 above)
Comment (channa-figure) [reply x/marker/keeper/querier.go:167]: Sorry, totally didn't notice that comment.

**Review by dwedul-figure — DISMISSED** [review]
LGTM

**Review by iramiller — DISMISSED** [review]
LGTM.

**Review by iramiller — APPROVED** [review]
👍🏻

**Review by dwedul-figure — APPROVED** [review]
LGTM

**Review by iramiller — APPROVED** [review]

**Pull request merged**